modules/auth/auth.controller.js

```diff
@@ -0,0 +1,81 @@
+const jwt = require("jsonwebtoken");
+const passportJwt = require("passport-jwt");
+const { User } = require("../users/user.schema");
+const { createUser } = require("../users/users.controller");
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+  const user = await User.findOne({ email });
+
+  if (!user || !user.validPassword(password)) {
+    return res.status(400).json({ message: "Hasło lub Email niepoprawne" });
+  }
+
+  const token = jwt.sign(
+    {
+      id: user._id,
+    },
+    "secret",
+    {
+      expiresIn: "1h",
+    }
+  );
+
+  user.token = token;
+  user.save();
+
+  return res.json({
+    data: { token },
+  });
+};
+
+const register = async (req, res) => {
+  if (
+    !Object.hasOwn(req.body, "email") ||
+    !Object.hasOwn(req.body, "password")
+  ) {
+    return res
+      .status(400)
+      .json({ message: "Błąd z Joi lub innej biblioteki walidacji" });
```

Comment:
Tu potrzebujemy błędu prawdziwego :)

Reply:
Trzeba by dodać bibliotekę joi  lub yup inwalidować pola które wymagają aby były obowiązkowe. Bo to powyższe sprawdzenie jest bardzo luźne :)

Reply:
Poprawiłam wiadomość